pkgs/top-level/all-packages.nix

```diff
@@ -17180,7 +17180,7 @@ with pkgs;
   };
 
   inherit (beam.interpreters)
-    erlang erlang_27-rc3 erlang_26 erlang_25 erlang_24
+    erlang erlang_27 erlang_26 erlang_25 erlang_24
```

Comment:
If you remove versions like that you'll be breaking `erlang_27-rc3` without notice for whoever is using this version.

Split the different objectives:
1) Add new erlang version (`erlang_27: init 27.x.x`)
2) Upgrade main erlang (`erlang: erlang_27-rc3 -> erlang_27`)
3) Remove erlang_27-rc3 (`erlang_27-rc3: remove`) => It should be an alias, to be turned into a warning on release and then to be removed after release. Minimally it should be a warning, such that the user gets some warning of what is going on.

Reply:
I have never turned these things into warnings. Do you have an example?

FWIW, the `rc{N}` versions were only added for testing purposes. I agree a warning would be ideal though.

Reply:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/aliases.nix#L338

Reply:
In retrospect, likely `erlang_27-rc3` should have been tagged to `erlang_latest` or `erlang_edge` (for pre-releases).

Reply:
Yeah, that is a good idea. Do you think it would be sufficient to just tag the warning for `erlang_27-rc3` and initialize `erlang_27` proper here? (again, this was my fault for not setting this up for success)

Reply:
Also, I'll fix the commit message tonight

Reply:
To remove the package, add an alias:

> erlang_27-rc3 = throw "erlang_27-rc3 has been removed in favor of newer versions."; # added 2024-05-20

To: https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/aliases.nix#L343

You need 3 commits: One for:
* adding the new erlang version.
* bumping the erlang tag
* removing the old version. (Which then creates the alias warning that such package is now unavailable.) I think this package should be removed because it is a release candidate, and it is not for an edge tag.

Reply:
Ahh, I thought you wanted me to split these into 3 PRs which felt a little tedious. Thanks for the clarification.

I'll get this cleaned up tonight.

Reply:
To be honest, the commit messages for the latter two I am unsure of the convention

- `erlang: init at 27.0`: seems correct
- `erlang: erlang_27-rc3 -> erlang_27`: seems okay, deprecate one in favor of another
- `erlang: erlang_25 -> erlang_27 (latest version)`: totally unsure, `erlang: latest version to 27.0`?

Reply:
> The following commit message doesn't follow nixpkgs guidelines on commit messages:
> 
> > erlang_27-rc3 -> erlang_27
> 
> When removing a package tag (erlang_27-rc3), add a warning (at aliases) for notifying users of such deprecation.
> 
> Split the different objectives:
> 
>     1. Add new erlang version (`erlang_27: init 27.x.x`)
> 
>     2. Upgrade main erlang (`erlang: erlang_27-rc3 -> erlang_27`)
> 
>     3. Remove erlang_27-rc3 (`erlang_27-rc3: remove`) => It should be an alias, to be turned into a warning on release and then to be removed after release. Minimally it should be a warning, such that the user gets some warning instead of a not found error.
> 
> 
> *Notice: The text inside parameters is the expected commit messages title.

Please, re-read this. ^

Reply:
I could also these as being acceptable?

```
erlang_27: init at 27.0
erlang: erlang_27-rc3 -> erlang_27
erlang: latestVersion -> erlang_27
```

Reply:
Please, make the commit removing version `erlang_27-rc3` a discrete commit.

It should be titled as:

> `erlang_27-rc3: remove`

Then, certify only changes related to the removal of package exist in this commit. Such as package remove and adding alias deprecation warning.

Reply:
Okay, I see my confusion now. In my brain, `erlang: erlang_27-rc3 -> erlang_27` is saying "replace erlang_27-rc3 with erlang_27" which implies I am adding the alias for `erlang_27-rc3` in the same commit (because it was replaced). You want it in a separate one (which is why I was thinking you meant `latestVersion`). Sorry, this is genuinely confusing to me.

Reply:
I guess we should also update the "latest version" though. How is that supposed to look?

Reply:
> Okay, I see my confusion now. (...) Sorry, this is genuinely confusing to me.

If you have patience, we'll figure out this. Communication is hard. The good side is, once you learn the pattern, reviews will be easier.